spring/data-core-jakarta/src/main/kotlin/com/linecorp/kotlinjdsl/spring/data/SpringDataQueryFactoryExtensions.kt

```diff
@@ -7,7 +7,7 @@ import java.util.stream.Stream
 
 inline fun <reified T> SpringDataQueryFactory.singleQuery(
     noinline dsl: SpringDataCriteriaQueryDsl<T>.() -> Unit
-): T = selectQuery(T::class.java, dsl).singleResult
+): T? = selectQuery(T::class.java, dsl).resultList.firstOrNull()
```

Comment:
The singleQuery method implements JPA's singleResult interface. Therefore, I think it would be better to make an exception if it is null instead of returning nullable.

If you want nullable, I think it would be better to create an additional extension like `singleOrNullQuery` method.

-- korean
singleQuery 메소드는 JPA의 singleResult 인터페이스를 구현하고 있습니다. 그렇기 때문에 nullable을 반환하는 것보다 null이면 예외를 발생시키는 것이 더 좋을 것으로 생각됩니다.

만약 예외 대신 nullable 반환을 원하시면 `singleOrNullQuery`확장 메소드를 만드는 게 더 좋을 것으로 보입니다.

Reply:
I also agree with @shouwn's comment. At the time of creating singleQuery, there was a part that tried to match the basic specifications of JPA as much as possible. However, as singleQueryOrNull exists in jdsl-reactive, which implements hibernate-reactive, it would be better to create an extension function for the existing version.

-- korean
저도 @shouwn 님의 의견과 일치합니다. singleQuery 를 만들었을 당시에는 JPA 의 기본 스펙과 최대한 일치하려고 했던 부분이 있습니다. 그러나 hibernate-reactive 를 구현한 jdsl-reactive 에서는 singleQueryOrNull 이 존재하는 만큼 기존 버전에도 확장함수를 만드는게 더 좋을 것 같습니다.